plugins/TouchIcon/class.touchicon.plugin.php

```diff
@@ -43,33 +60,45 @@ public function SettingsController_TouchIcon_Create($Sender) {
       $Sender->Permission('Garden.Settings.Manage');
       $Sender->AddSideMenu('dashboard/settings/touchicon');
       $Sender->Title(T('Touch Icon'));
-      
+
       if ($Sender->Form->AuthenticatedPostBack()) {
          $Upload = new Gdn_UploadImage();
          try {
             // Validate the upload
             $TmpImage = $Upload->ValidateUpload('TouchIcon', FALSE);
             if ($TmpImage) {
-               // Save the uploaded image
+               // Save the uploaded image.
+               $TouchIconPath ='banner/touchicon_'.substr(md5(microtime()), 16).'.png';
                $Upload->SaveImageAs(
                   $TmpImage,
-                  PATH_ROOT.'/uploads/TouchIcon/apple-touch-icon.png',
+                  $TouchIconPath,
                   114,
                   114,
                   array('OutputType' => 'png', 'ImageQuality' => '8')
                );
-               SaveToConfig('Plugins.TouchIcon.Uploaded', TRUE);
+               SaveToConfig('Garden.TouchIcon', $TouchIconPath);
```

Comment:
The `$Upload->SaveImageAs()` method returns an array with a `SavePath` key (I think). That should be used in the config instead.